spring-integration/spring-boot-autoconfigure/src/main/java/ai/timefold/solver/spring/boot/autoconfigure/TimefoldAutoConfiguration.java

```diff
@@ -64,62 +70,422 @@
+                    .filter(e -> e.getValue().getSolutionClass() == null)
+                    .map(Map.Entry::getKey)
+                    .toList();
+            if (annotationInstanceSet.size() > 1 && !solverConfigWithoutSolutionClassList.isEmpty()) {
+                throw new IllegalStateException(
+                        """
+                                Some solver configs (%s) don't specify a %s class, yet there are multiple available (%s) on the classpath.
+                                Maybe set the XML config file to the related solver configs, or add the missing solution classes to the XML files,
+                                or remove the unnecessary solution classes from the classpath."""
+                                .formatted(String.join(", ", solverConfigWithoutSolutionClassList),
+                                        PlanningSolution.class.getSimpleName(),
+                                        annotationInstanceSet.stream().map(Class::getSimpleName).collect(joining(", "))));
+            }
+            // Unused solution classes
+            List<String> unusedSolutionClassList = annotationInstanceSet.stream()
+                    .map(Class::getName)
+                    .filter(planningClassName -> solverConfigMap.values().stream().filter(c -> c.getSolutionClass() != null)
+                            .noneMatch(c -> c.getSolutionClass().getName().equals(planningClassName)))
+                    .toList();
+            if (annotationInstanceSet.size() > 1 && !unusedSolutionClassList.isEmpty()) {
+                throw new IllegalStateException(
+                        "Unused classes ([%s]) found with a @%s annotation.".formatted(
+                                String.join(", ", unusedSolutionClassList),
+                                PlanningSolution.class.getSimpleName()));
+            }
+            // TODO - Should we validate target types?
+        } catch (ClassNotFoundException e) {
+            throw new IllegalStateException(
+                    "Scanning for @%s annotations failed.".formatted(PlanningSolution.class.getSimpleName()), e);
+        }
+    }
+
+    private void assertSolverConfigEntityClasses(IncludeAbstractClassesEntityScanner entityScanner) {
+        // No Entity class
+        String emptyListErrorMessage = """
+                No classes were found with a @%s annotation.
+                Maybe your @%s annotated class(es) are not in a subpackage of your @%s annotated class's package.
+                Maybe move your planning entity classes to your application class's (sub)package(or use @%s)."""
+                .formatted(
+                        PlanningEntity.class.getSimpleName(), PlanningEntity.class.getSimpleName(),
+                        SpringBootApplication.class.getSimpleName(), EntityScan.class.getSimpleName());
+        assertEmptyInstances(entityScanner, PlanningEntity.class, emptyListErrorMessage);
+        // TODO - Should we validate target types?
```

Comment:
Let's not leave the TODO here, let's fix it instead.

What would be the pros and cons?
What does Quarkus do?

Reply:
Quarkus logic requires that all planning entities be implemented as Java classes.

Reply:
Let me implement the same logic for Spring boot